### DISCUSSION TITLE
Input file for OperFlag=3 in "examples/Wind-Turbine/NREL-5MW_ALM"

### DISCUSSION THREAD

**Discussion by hamidMech** [xcompact3d/Incompact3d#231] [Q&A] [answered]
Hello,
How to modify "NREL-5MW.turb" input file in the "examples/Wind-Turbine/NREL-5MW_ALM" to read the pitch angle of each blade as a function of time from a file like "pitch.in"? How would it be the format of "pitch.in"?
Best,
Hamid

**Comment by nbeb** [comment, marked as answer]
Hi Hamid, 

If you want to change the pitch of the blades (either collectively or individually) as a function of time while keeping the rotational speed of the turbine constant, you would need to do some minor modifications in the code. In particular, in acl_model.f90, you would need to add the pitch control routines (e.g., lines 547-557) in the Is_constant_rotation_operated condition (line 490). The pitch can then be expressed as a function of time, as for example is done in line 578. Alternatively, to read this info from a file, you would need to add this capability yourself, 

Best, 
Nikos
